src/components/TeamMembers/TeamMembers.tsx

```diff
@@ -0,0 +1,240 @@
+import { IconButton } from "../IconButton";
+import { Modal } from "../Modal";
+import MultiSelectList from "../MultiSelectList/MultiSelectList";
+import { toastError, toastSuccess } from "../Toast/toast";
+import { useLoader } from "../../hooks";
+
+type TeamMembersProps = {
+  teamId: string;
+};
+
+export function TeamMembers({ teamId }: TeamMembersProps) {
+  const [teamMembers, setTeamMembers] = useState<User[]>([]);
+  const [allTeamMembers, setAllTeamMembers] = useState<User[]>([]);
+  const [selectedMembers, setSelectedMembers] = useState<User[]>([]);
+  const [open, setOpen] = useState<boolean>(false);
+  const { loading, setLoading } = useLoader();
+
+  async function fetchTeamMembers(id: string) {
+    setLoading(true);
+    const { data } = await getTeamMembers(id);
+    try {
+      const { data: users, error } = await getPersons();
+      setTeamMembers(data || []);
+      setAllTeamMembers(
+        (users || []).filter(
+          (user) => !data?.find((item) => item.id === user.id)
+        )
+      );
+      if (error) {
+        toastError(error.message);
+      }
+    } catch (ex) {
+      toastError(ex);
+    }
+    setLoading(false);
+  }
+
+  async function removeTeamMember(userId: string) {
+    try {
+      const { error } = await removeUserFromTeam(userId);
+      if (!error) {
+        toastSuccess("user removed from team successfully");
+        fetchTeamMembers(teamId);
+      } else {
+        toastError(error.message);
+      }
+    } catch (ex) {
+      toastError(ex);
+    }
+  }
+
+  async function addTeamMembers(selectedUsers: User[]) {
+    try {
+      const { error } = await addUserToTeam(
+        teamId,
+        selectedUsers.map((item) => item.id)
+      );
+      if (!error) {
+        toastSuccess("team members added successfully");
+        fetchTeamMembers(teamId);
+      } else {
+        toastError(error.message);
+      }
+    } catch (ex) {
+      toastError(ex);
+    }
+    setOpen(false);
+  }
+
+  useEffect(() => {
+    if (!teamId) {
+      return;
+    }
+    fetchTeamMembers(teamId);
+  }, [teamId]);
+
+  return (
+    <div className="p-4 pb-8 space-y-2 bg-white rounded shadow">
+      <h2 className="text-dark-gray font-bold mr-3 text-xl flex items-center space-x-2">
+        Team Members
+      </h2>
+      <div className="flex flex-col">
+        <div className="text-center">
+          <FaUsers className="mx-auto h-12 w-12 text-gray-400" />
+          <h2 className="mt-2 text-lg font-medium text-gray-900">
+            Add team members
+          </h2>
+          {!teamMembers.length && !loading && (
+            <p className="mt-1 text-sm text-gray-500">
+              You haven't added any team members to the team yet.
+            </p>
+          )}
+          <button
+            type="submit"
+            className="btn-primary mt-4"
+            onClick={() => setOpen(true)}
+          >
+            Add Members
+          </button>
+        </div>
+      </div>
```

Comment:
This block seems a bit repetitive - Should this use an EmptyState component (Maybe with the addition of the icon) ?